lib/castle/header_filter.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module Castle
+  # used for preparing valuable headers list
+  class HeaderFilter
+    # headers filter
```

Comment:
Can we add a comment why these are valuable so that the next person coming understands this?

```
HTTP_ - this is how Rack prefixes incoming HTTP headers
CONTENT_LENGTH - it's here for legacy reasons, may be not really needed
REMOTE_ADDR - ??? 
```